cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -615,8 +616,104 @@ SwiftGenerator::writeServantDocSummary(IceInternal::Output& out, const Interface
 void
 SwiftGenerator::validateMetadata(const UnitPtr& u)
 {
-    MetadataVisitor visitor;
-    u->visit(&visitor);
+    map<string, MetadataInfo> knownMetadata;
+
+    // "swift:attribute"
+    MetadataInfo attributeInfo = {
+        .validOn = {typeid(ClassDecl), typeid(Struct), typeid(Slice::Exception), typeid(Enum)},
+        .acceptedArgumentKind = MetadataArgumentKind::RequiredTextArgument,
+        .mustBeUnique = false,
+    };
+    knownMetadata.emplace("swift:attribute", attributeInfo);
+
+    // "swift:class-resolver-prefix"
+    MetadataInfo classResolverPrefixInfo = {
+        .validOn = {typeid(Unit)},
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
+    };
+    knownMetadata.emplace("swift:class-resolver-prefix", classResolverPrefixInfo);
+
+    // "swift:inherits"
+    MetadataInfo inheritsInfo = {
+        .validOn = {typeid(InterfaceDecl)},
+        .acceptedArgumentKind = MetadataArgumentKind::RequiredTextArgument,
+        .mustBeUnique = false,
+    };
+    knownMetadata.emplace("swift:inherits", inheritsInfo);
+
+    // "swift:module"
+    MetadataInfo moduleInfo = {
+        .validOn = {typeid(Module)},
+        // Even though it's really 'module:prefix' the validator sees this as a single argument since there's no commas.
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
+    };
+    knownMetadata.emplace("swift:module", moduleInfo);
+
+    // Pass this information off to the parser's metadata validation logic.
+    Slice::validateMetadata(u, "swift", knownMetadata);
+}
+
+void
+SwiftGenerator::validateSwiftModuleMappings(const UnitPtr& unit)
```

Comment:
All this code was just moved out of `MetadataValidator::visitUnit` into a new static function.
I didn't change anything else about it.